sympy/core/relational.py

```diff
@@ -1138,3 +1047,309 @@ def _eval_relation(cls, lhs, rhs):
+    >>> is_lt(_sympify(2), _sympify(0))
+    False
+
+   """
+    if not (isinstance(lhs, Expr) and isinstance(rhs, Expr)):
+        raise TypeError("Can only compare inequalities with Expr")
+
+    retval = _eval_is_ge(lhs, rhs)
+
+    if retval is not None:
+        return retval
+    else:
+        n2 = _n2(lhs, rhs)
+        if n2 is not None:
+            # use float comparison for infinity.
+            # otherwise get stuck in infinite recursion
+            if n2 in (S.Infinity, S.NegativeInfinity):
+                n2 = float(n2)
+            return _sympify(n2 >= 0)
+        if lhs.is_extended_real and rhs.is_extended_real:
+            if (lhs.is_infinite and lhs.is_extended_positive) \
+                or (rhs.is_infinite and rhs.is_extended_negative):
+                return True
+            diff = lhs - rhs
+            if diff is not S.NaN:
+                rv = diff.is_extended_nonnegative
+                if rv is not None:
+                    return rv
+
+
+def is_neq(lhs, rhs):
+    """Fuzzy bool for lhs does not equal rhs.
+
+    See the docstring for is_eq for more
+    """
+    return fuzzy_not(is_eq(lhs, rhs))
+
+
+def is_eq(lhs, rhs):
+    """
+    Fuzzy boolean used to check if lhs is equal, mathematically
+    to rhs.
+    Returns True or False if the two are comparable, or None if the two are incomparable
+    or if the comparison is indeterminate.
+
+     To override the default equality logic, use _eval_is_eq with multiple dispatch.
+     is_ne, fuzzy nots is_eq so, to override the logic of is_ne, you will need to override _eval_is_eq
+
+    Examples
+    ========
+
+    >>> from sympy.core.relational import is_eq
+    >>> from sympy.core.sympify import _sympify
+    >>> is_eq(_sympify(0), _sympify(0))
+    True
+
+    >>> from sympy.core.relational import is_neq
+    >>> from sympy.core.sympify import _sympify
+    >>> is_neq(_sympify(0), _sympify(0))
+    False
+
+    >>> from sympy.core.relational import is_eq
+    >>> from sympy.core.sympify import _sympify
+    >>> is_eq(_sympify(0), _sympify(2))
+    False
+
+    >>> from sympy.core.relational import is_neq
+    >>> from sympy.core.sympify import _sympify
+    >>> is_neq(_sympify(0), _sympify(2))
+    True
+
+    >>> from sympy.abc import x
+    >>> from sympy.core.relational import is_eq
+    >>> from sympy.core.sympify import _sympify
+    >>> is_eq(_sympify(0), x)
+
+    """
+    from sympy.core.add import Add
+    from sympy.functions.elementary.complexes import arg
+    from sympy.simplify.simplify import clear_coefficients
+    from sympy.utilities.iterables import sift
+
+    retval = _eval_is_eq(lhs, rhs)
+    if retval is not None:
+        return retval
+
+    if dispatch(type(lhs), type(rhs)) != dispatch(type(rhs), type(lhs)):
+        retval = _eval_is_eq(rhs, lhs)
+        if retval is not None:
+            return retval
+
+    eval_func = getattr(lhs, '_eval_Eq', None)
+    if eval_func is not None:
```

Comment:
There should be a comment here explaining that this method is being called for backwards compatibility and that it is better to use `_eval_is_eq` instead.